docs/architecture.md

```diff
@@ -0,0 +1,25 @@
+# Architecture
+
+## The chosen architecture
+libcal-proxy is a [vert.x](https://vertx.io/) application.
+It consists of:
+* A verticle that 
+** Loads application configuration
+** Retrieves initial OAuth tokens
+** Deploys the two request handlers
+* A status handler, called to determine if the application is running
+* A LibCal Proxy handler, which
+** Receives LibCal API requests
+** Retrieves an OAuth token from application storage
+** Passes the API request along to LibCal with the token
+** Returns the LibCal response to the original client
+
+## Functional Overview
+The proxy exists to insulate cients from the process of retrieving an OAuth token from the LibCal OAuth provider. Client calls are identical to direct calls to a LibCal API, except for two points: 
+1) "proxy" is prefixed to the UCLA LibCal URL (e.g.: if the direct LibCal call is https://calendar.library.ucla.edu/1.1/event_search?search=*, the proxy call would be https://proxy.calendar.library.ucla.edu/1.1/event_search?search=*)
```

Comment:
Not sure why the second half of this is italicized(?)

Reply:
Apparently the "*" in the sample URLs was being interpreted as styling.

Reply:
Ah! I always forget *s are italics in markdown (in addition to underscores).